File: src/main/java/com/nndmove/app/service/dto/GenresDTO.java
package com.nndmove.app.service.dto;

import jakarta.validation.constraints.*;
import java.io.Serializable;
import java.util.HashSet;
import java.util.Objects;
import java.util.Set;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * A DTO for the {@link com.nndmove.app.domain.Genres} entity.
 */
@Getter
@Setter
@NoArgsConstructor
@SuppressWarnings("common-java:DuplicatedBlocks")
public class GenresDTO implements Serializable {

    private Long id;

    @NotNull
    private String genres;

    private Set<MovieDTO> movies = new HashSet<>();

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof GenresDTO)) {
            return false;
        }

        GenresDTO genresDTO = (GenresDTO) o;
        if (this.id == null) {
            return false;
        }
        return Objects.equals(this.id, genresDTO.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.id);
    }

    // prettier-ignore
    @Override
    public String toString() {
        return "GenresDTO{" +
            "id=" + getId() +
            ", genres='" + getGenres() + "'" +
            ", movies=" + getMovies() +
            "}";
    }
}

File: src/main/java/com/nndmove/app/service/dto/GenresDTO.kt
package com.nndmove.app.service.dto

import jakarta.validation.constraints.NotNull
import java.io.Serializable
import java.util.*

/**
 * A DTO for the [com.nndmove.app.domain.Genres] entity.
 */
class GenresDTO : Serializable {
    @JvmField
    var id: Long? = null

    @JvmField
    var genres: @NotNull String? = null

    @JvmField
    var movies: Set<MovieDTO> = HashSet()

    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o !is GenresDTO) {
            return false
        }

        if (this.id == null) {
            return false
        }
        return this.id == o.id
    }

    override fun hashCode(): Int {
        return Objects.hash(this.id)
    }

    // prettier-ignore
    override fun toString(): String {
        return "GenresDTO{" +
                "id=" + id +
                ", genres='" + genres + "'" +
                ", movies=" + movies +
                "}"
    }
}
